File: lang/kotlin/src/main/resources/fr/epistudio/epysia/lang/kotlin/templates/Behaviour.kt
import fr.epistudio.epysia.EngineServices
import fr.epistudio.epysia.components.EpysiaComponent
import fr.epistudio.epysia.components.Export
import fr.epistudio.epysia.input.InputState
import fr.epistudio.epysia.scripting.Behaviour

@EpysiaComponent(name = "{{ClassName}}", category = "Scripts")
class {{ClassName}} : Behaviour() {

    @field:Export(label = "Speed")
    private var speed = 1.0f

    override fun onStart(services: EngineServices) {
    }

    override fun onUpdate(input: InputState, deltaTimeSeconds: Float) {
    }
}

File: lang/python/src/main/resources/fr/epistudio/epysia/lang/python/templates/Behaviour.py
from epysia import Behaviour, component, export


@component(name="{{ClassName}}", category="Scripts")
class {{ClassName}}(Behaviour):

    speed = export(1.0, label="Speed")

    def on_start(self):
        pass

    def on_update(self, input, delta_seconds):
        pass
